### PR TITLE
Bugfix/phone number may contain non ints

### LINKED ISSUE #27: bug: duplicate clients in db
Issues:
- `country_code` & `phone_number` are not unique in db
- We don't check/sanitize `country_code` & `phone_number` before inserting to db. This means `123-45` can be inserted along with `12345`. `country_code` & `phone_number` should be enforced to only contain digits

### PR DESCRIPTION
**Pull request by cglotr**
Fixes #27 

Changes:
- `country_code` & `phone_number` are unique in DB
- Added `getDigits` util func
- Added backend's first test! :100: 
- Make API call after client is successfully created on DB. This will save us from making unnecessary calls
- Error out in `main.go` so we can run tests on _Travis_

### CODE REVIEW

**Comment by cglotr** [comment]
@imranariffin Finally, backend's first ever test! :100: :1st_place_medal: ahahaha

File: db.go [hunk 1]
```diff
@@ -42,10 +42,11 @@ func init() {
 	db.Exec(`
 		CREATE TABLE IF NOT EXISTS client (
 			id INT UNSIGNED NOT NULL AUTO_INCREMENT,
-			country_code VARCHAR (50) NOT NULL,
-			phone_number VARCHAR (50) NOT NULL,
-			verification_code VARCHAR (50),
-			PRIMARY KEY (id)
+			country_code VARCHAR(50) NOT NULL,
```
Comment (imranariffin) [inline db.go]: country_code VARCHAR (50) NOT NULL, // consistent spacing

File: db.go [hunk 2]
```diff
@@ -52,6 +52,14 @@ func init() {
 }
 
 func getOrCreateClient(countryCode, phoneNumber string) (*Client, error) {
+	if countryCode != getDigits(countryCode) {
+		return nil, errors.New("hey, countryCode should only contain digits")
+	}
+
+	if phoneNumber != getDigits(phoneNumber) {
```
Comment (imranariffin) [inline db.go:61]: This will error when `phoneNumber` = "(647) 1234567", which is intended, right?

**Review by imranariffin — CHANGES_REQUESTED** [review]
Just a very minor code style change request & a question for clarification.

File: utils_test.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package main
+
+import (
+	"testing"
+)
+
+func TestGetDigits(t *testing.T) {
```
Comment (imranariffin) [inline utils_test.go:7]: Yay

File: db.go (hunk 2 above)
Comment (imranariffin) [reply db.go:61]: OK, I just read https://github.com/hooligram/hooligram-server/issues/27
Question resolved.

File: db.go (hunk 1 above)
Comment (cglotr) [reply db.go]: I was looking at [MySQL](https://dev.mysql.com/doc/refman/5.7/en/char.html) doc & saw that they write with no spacing for data types. But, now I think their spacing is kinda arbitrary too.

I agreed on consistent spacing 0b3dc1a43a5cfc2af5851c908f42405a20dbb103.

**Comment by cglotr** [comment]
> Let's merge

It's merged.

**Pull request merged**